### ISSUE TITLE
[Bug]: Unable to install specific version of git

### ISSUE THREAD

**Issue by tordenflesk** [ScoopInstaller/Main#6238]
### Prerequisites

- [X] I have written a descriptive issue title.
- [X] I have searched all issues/PRs to ensure it has not already been reported or fixed.
- [X] I have verified that I am using the latest version of Scoop and corresponding bucket.

### Package Name

git

### Expected/Current Behaviour

Scoop install git@2.46.2 results in:
```
WARN  Given version (2.46.2) does not match manifest (2.47.0)
WARN  Attempting to generate manifest for 'git' (2.46.2)
Autoupdating git
Searching hash for PortableGit-2.46.2-32-bit.7z.exe in https://github.com/git-for-windows/git/releases/tag/v$matchTag
The request was aborted: The connection was closed unexpectedly.
URL https://github.com/git-for-windows/git/releases/tag/v$matchTag is not valid
Could not find hash in https://github.com/git-for-windows/git/releases/tag/v$matchTag
Downloading PortableGit-2.46.2-32-bit.7z.exe to compute hashes!
The request was aborted: The connection was closed unexpectedly.
URL https://github.com/git-for-windows/git/releases/download/v$matchTag/PortableGit-2.46.2-32-bit.7z.exe#/dl.7z is not valid
Could not install git@2.46.2
Select-CurrentVersion : Cannot bind argument to parameter 'AppName' because it is an empty string.
At C:\Users\Tordenflesk\scoop\apps\scoop\current\lib\core.ps1:435 char:54
+     return $null -ne (Select-CurrentVersion -AppName $app -Global:$gl ...
+                                                      ~~~~
    + CategoryInfo          : InvalidData: (:) [Select-CurrentVersion], ParameterBindingValidationException
    + FullyQualifiedErrorId : ParameterArgumentValidationErrorEmptyStringNotAllowed,Select-CurrentVersion

You cannot call a method on a null-valued expression.
At C:\Users\Tordenflesk\scoop\apps\scoop\current\lib\manifest.ps1:36 char:5
+     $app = $app.TrimStart('/')
+     ~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidOperation: (:) [], RuntimeException
    + FullyQualifiedErrorId : InvokeMethodOnNull

Test-Path : Cannot bind argument to parameter 'Path' because it is an empty string.
At C:\Users\Tordenflesk\scoop\apps\scoop\current\lib\manifest.ps1:57 char:27
+             if (Test-Path $app) {
+                           ~~~~
    + CategoryInfo          : InvalidData: (:) [Test-Path], ParameterBindingValidationException
    + FullyQualifiedErrorId : ParameterArgumentValidationErrorEmptyStringNotAllowed,Microsoft.PowerShell.Commands.TestPathCommand

Couldn't find manifest for ''.
```

### Steps to Reproduce

```console
See above
```

### Possible Solution

No...

### Scoop and Buckets Version

```console
Current Scoop version:
v0.5.2 - Released at 2024-07-26
```

### Scoop Config

```console
scoop_branch          : master
last_update           : 2024-10-08T16:42:27.0889999+02:00
scoop_repo            : https://github.com/ScoopInstaller/Scoop
aria2-warning-enabled : False
aria2-enabled         : True
```

### PowerShell Version

```console
PSVersion                      5.1.14409.1018
PSEdition                      Desktop
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}
BuildVersion                   10.0.14409.1018
CLRVersion                     4.0.30319.42000
WSManStackVersion              3.0
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
```

**Comment by HUMORCE** [comment]
It's broken after autoupdate fixes since version 2.41.0.2

Auto-generation of manifests is not a 100% reliable feature. For this case, copy the raw file from the commit history then install it.

**Comment by JatinSanghvi** [comment]
In your case, the following command should work.

```
scoop uninstall git
scoop install https://raw.githubusercontent.com/ScoopInstaller/Main/6ef718a96949c6d1aece63b6cea83d3aedc7125d/bucket/git.json
```

The above URL belongs to Git version 2.46.2 and will change if a different Git version needs to be installed.
